### PR TITLE
cleanup: Cleaned-up protobufs appendix and add some missing refs to it

### LINKED ISSUE #67: protobuf RPC API - remove duplicated definitions in appendix
With #24 we introduced a description of the gRPC CP API in the appendix.  However, with this, some definitions are duplicated. For example, `ASEntrySignedBody` is defined in both `2.2.1.4.2.  AS Entry Signed Body` and in the appendix. 
 
TODO: 
- [x] evaluate wether we leave the gRPC API descriptions all in the appendix or in various paragraphs --> we leave them in both places
- [x] ~some gRPC definitions are duplicated, duplicates should be removed (preferably, leave them in the respective chapters rather than in appendix). A non-exhaustive list:~
  - `SegmentType`
  - `SegmentsRegistrationRequest`
  - `SegmentCreationService`
  - `BeaconRequest`
  - `BeaconResponse`
  - `PathSegment`
  - `SegmentInformation`
  - `ASEntry`
  - `ASEntrySignedBody`
  - `HopEntry`
  - `PeerEntry`
  - `HopField`
  - `SignedMessage`
  - `Header`
  - `HeaderAndBodyInternal`
  - `VerificationKeyID`
- [x] make sure there are proper cross-references to the respective gRPC requests For example: 
  - `SegmentsRequest` right now it is not clear which requests are used for path lookups by endpoints, e.g. we should mention that an endpoint sends a `SegmentsRequest` to its path server.
  - `SegmentsResponse`
  - `SignatureAlgorithm`
- [x] We sometimes call the API descritpions in the text "on Code-Level", we should use a better wording (e.g. gRPC API)
- [x] check descriptions and typos  (e.g. `// This API is exposed by the control services of core ASes expose this on the SCION dataplane and also by all`

### PR DESCRIPTION
**Pull request by jiceatscion**
Also cleaned-up extraneous blank lines. Replaced phrases like "on xyz-level".

Fixes #67

### CODE REVIEW

File: draft-dekater-scion-controlplane.md [hunk 1]
```diff
@@ -1711,18 +1640,32 @@ message SegmentsRequest {
...
     }
 
     // Mapping from path segment type to path segments.
-    // The key is the integer
-    // representation of the SegmentType enum.
+    // The key is the integer representation of the SegmentType enum.
     map<int32, Segments> segments = 1;
 }
+~~~~
+{: #figure-11 title="Control Service gRPC API - Segment lookup"}
```
Comment (nicorusti) [inline draft-dekater-scion-controlplane.md:1665]:
```suggestion
{: #figure-11 title="Control Service gRPC API - Segment lookup. This API is exposed on the SCION dataplane by the control services of core ASes and exposed on the "intra-domain protocol" network."}
```

File: draft-dekater-scion-controlplane.md [hunk 2]
```diff
@@ -1711,18 +1640,32 @@ message SegmentsRequest {
...
+<br>
 
+~~~~
 // This API is only exposed by core ASes and only on the SCION
 // dataplane.
```
Comment (nicorusti) [inline draft-dekater-scion-controlplane.md:1670]: Same as in comment above

**Review by nicorusti — APPROVED** [review]
Thanks a lot! I left some minor comments/changes, the rest looks good!

File: draft-dekater-scion-controlplane.md [hunk 3]
```diff
@@ -1668,37 +1613,21 @@ This document has no IANA actions.
...
-    // Core segment.
-    SEGMENT_TYPE_CORE = 3;
-}
-
-
-// This API is exposed by the control services of core ASes expose this
-// on the SCION dataplane and also by all control services on the
-// "intra-domain protocol" network.
+// This API is exposed on the SCION dataplane by the control services
+// of core ASes and exposed on the "intra-domain protocol" network.
```
[the remark this answers is not in the document]
Comment (jiceatscion) [reply draft-dekater-scion-controlplane.md]: done

File: draft-dekater-scion-controlplane.md (hunk 1 above)
Comment (jiceatscion) [reply draft-dekater-scion-controlplane.md]: done

**Pull request merged**